### PR TITLE
Dockerfile updates for traject

### PR DESCRIPTION
**Pull request by aelkiss**
Brings our dockerfile up to current standards & allows building an image we can use in kubernetes for catalog indexing

* remove wait-for & use health checks
* use production stage

### CODE REVIEW

File: lib/translation_maps/ht/collection_code_to_original_from.yaml
Comment (aelkiss) [inline lib/translation_maps/ht/collection_code_to_original_from.yaml:1]: should revert this change

**Review by Ronster2018 — COMMENTED** [review]
These changes look good and make sense to me.

File: lib/translation_maps/ht/collection_code_to_original_from.yaml
Comment (moseshll) [reply lib/translation_maps/ht/collection_code_to_original_from.yaml:1]: I wouldn't be opposed to bringing this file up to date with whatever is in the production DB and then `.gitignore`ing it. I've run against problems using this repo (or maybe it was `hathifiles`) with a stale table like this, against real-world Zephir data (bails out due to inability to find a recent institution code). It'll always be a moving target, however. Maybe worth a note in the readme.

**Review by moseshll — APPROVED** [review]
Looks good and tests good. I commented on the collection code file, no resolution to that will be perfect.

File: lib/translation_maps/ht/collection_code_to_original_from.yaml
Comment (aelkiss) [reply lib/translation_maps/ht/collection_code_to_original_from.yaml:1]: @moseshll If you have time to do that then we can go ahead and merge this.

**Review by Ronster2018 — APPROVED** [review]

File: lib/translation_maps/ht/collection_code_to_original_from.yaml
Comment (moseshll) [reply lib/translation_maps/ht/collection_code_to_original_from.yaml:1]: Updated from production DB, added a README suggestion on how to ignore this file using `git update-index --skip-worktree`
